omnigibson/robots/behaviorbot.py

```diff
@@ -0,0 +1,677 @@
+                # controller mode, map trigger fraction from [0, 1] to [-1, 1] range.
+                action[self.controller_action_idx[f"gripper_{part_name}"]] = hand_data * 2 - 1
+            elif hand_data is not None:
+                # hand tracking mode, compute joint rotations for each independent hand joint
+                action[self.controller_action_idx[f"gripper_{part_name}"]] = hand_data[:, :2].T.reshape(-1)
+            # If we reset, teleop the robot parts to the desired pose
+            if reset:
+                self.parts[part_name].set_position_orientation(des_local_part_pos, des_part_rpy)
+            # update ghost hand if necessary
+            if self.use_ghost_hands and part_name is not "head":
+                self.parts[part_name].update_ghost_hands(des_world_part_pos, des_world_part_orn)
+        return action
+
+
+class BRPart(ABC):
+    """This is the interface that all Behaviorbot eef parts must implement."""
+
+    def __init__(self, name: str, parent: Behaviorbot, prim_path: str, eef_type: str, offset_to_body: List[float]) -> None:
+        """
+        Create an object instance with the minimum information of class ID and rendering parameters.
+
+        Args:
+            name (str): unique name of this BR part
+            parent (Behaviorbot): the parent BR object
+            prim_path (str): prim path to the root link of the eef
+            eef_type (str): type of eef. One of hand, head
+            offset_to_body (List[float]): relative rotation offset between the shoulder_rz link and the eef link.
+        """
+        self.name = name
+        self.parent = parent
+        self.prim_path = prim_path
+        self.eef_type = eef_type
+        self.offset_to_body = offset_to_body
+
+        self.ghost_hand = None
+        self._root_link = None
+        self._world_position_orientation = ([0, 0, 0], [0, 0, 0, 1])
+
+    def load(self) -> None:
+        self._root_link = self.parent.links[self.prim_path]
+        # setup ghost hand
+        if self.eef_type == "hand" and self.parent.use_ghost_hands:
+            gh_name = f"ghost_hand_{self.name}"
+            self.ghost_hand = USDObject(
+                prim_path=f"/World/{gh_name}",
+                usd_path=os.path.join(gm.ASSET_PATH, f"models/behaviorbot/usd/{gh_name}.usd"),
+                name=gh_name,
+                scale=0.001,
+                visible=False,
+                visual_only=True,
+            )
+            og.sim.import_object(self.ghost_hand)
+
+    @property
+    def local_position_orientation(self) -> Tuple[Iterable[float], Iterable[float]]:
+        """
+        Get local position and orientation w.r.t. to the body
+        Returns:
+            Tuple[Array[x, y, z], Array[x, y, z, w]]
+
+        """
+        return T.relative_pose_transform(*self.world_position_orientation, *self.parent.get_position_orientation())
+
+    @property
+    def world_position_orientation(self) -> Tuple[Iterable[float], Iterable[float]]:
```

Comment:
maybe redundant? or rename to get_position_orientation() to mirror set_position_orientation?

Reply:
renamed